### PR TITLE
499-be-bug-seen-and-not-seen-not-implemented-not-working-properly

### PR DESCRIPTION
**Pull request by xpan1c**
- Removed status from zod and Prisma schemas
- Added POST /seen/:resourceId
- Updated GET /resources controller
- Added param zod schema

### CODE REVIEW

File: back/src/__tests__/resources/get.test.ts [hunk 1]
```diff
@@ -166,15 +177,50 @@ describe('Testing resources GET endpoint', () => {
...
       .get(`${pathRoot.v1.resources}`)
       .query({ status: 'SEEN' })
     expect(response.status).toBe(200)
-    expect(response.body.length).toBeGreaterThanOrEqual(1)
+    expect(response.body.length).toBe(createdResources.length)
+    response.body.forEach((resource: ResourceGetSchema) => {
+      expect(() => resourceGetSchema.parse(resource)).not.toThrow()
+    })
+  })
+  it('should get viewed resources with status SEEN when logged in', async () => {
```
Comment (kevinmamaqi) [inline back/src/__tests__/resources/get.test.ts:190]: It should get all resources, without status in the response when not logged in

File: back/src/__tests__/resources/get.test.ts [hunk 2]
```diff
@@ -166,15 +177,50 @@ describe('Testing resources GET endpoint', () => {
...
+      .set('Cookie', authToken.admin)
+      .query({ status: 'SEEN' })
+    expect(response.status).toBe(200)
+    expect(response.body.length).toBe(1)
+    expect(viewedResource.resourceId).toBe(createdResources[0].id)
+    response.body.forEach((resource: ResourceGetSchema) => {
+      expect(() => resourceGetSchema.parse(resource)).not.toThrow()
+    })
+  })
+  it('should get all resources with status NOT_SEEN when not logged in', async () => {
```
Comment (kevinmamaqi) [inline back/src/__tests__/resources/get.test.ts:202]: It should get all resources, without status in the response when not logged in

File: back/src/__tests__/resources/get.test.ts [hunk 3]
```diff
@@ -166,15 +177,50 @@ describe('Testing resources GET endpoint', () => {
...
+    const response = await supertest(server)
+      .get(`${pathRoot.v1.resources}`)
+      .query({ status: 'NOT_SEEN' })
+    expect(response.status).toBe(200)
+    expect(response.body.length).toBe(createdResources.length)
+    response.body.forEach((resource: ResourceGetSchema) => {
+      expect(() => resourceGetSchema.parse(resource)).not.toThrow()
+    })
+  })
+  it('should get all resources with status NOT_SEEN when logged in', async () => {
```
Comment (kevinmamaqi) [inline back/src/__tests__/resources/get.test.ts:212]: Aquí testear que filtra SEEN/NOT_SEEN

File: back/src/__tests__/resources/get.test.ts (hunk 2 above)
Comment (kevinmamaqi) [reply back/src/__tests__/resources/get.test.ts:202]: Cuando no esta logeado no hace falta el status

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

File: back/src/controllers/seen/create.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  if (!resourceId) throw new MissingParamError('resourceId')
+  const resourceFound = await prisma.resource.findUnique({
+    where: {
+      id: resourceId,
+    },
+  })
+
+  if (!resourceFound) throw new NotFoundError('Resource not found')
+
+  const existingViewedResource = await prisma.viewedResource.findUnique({
```
Comment (kevinmamaqi) [inline back/src/controllers/seen/create.ts:19]: use primsa upsert

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

**Review by alex0rpi — APPROVED** [review]

**Review by kevinmamaqi — APPROVED** [review]

**Pull request merged**